helpers/validators.js

```diff
@@ -0,0 +1,36 @@
+const path = require('path');
+const fs = require('fs');
+
+const reIdentifier = /^[_a-zA-Z][_a-zA-Z0-9]*$/;
+const reVersion = /^\d+\.\d+\.\d+$/;
+
+module.exports = {
+    validatePath(value) {
+        const resolved = path.resolve(value);
+        if (!fs.existsSync(resolved)) {
+            return `${resolved} doesn't seem to exist!`;
+        }
+        return true;
+    },
+
+    validateIdentifier(value) {
+        if (!reIdentifier.test(value)) {
+            return `${value} is not a valid identifier. It must follow the format ${reIdentifier.source}`;
+        }
+        return true;
+    },
+
+    validateVersion(value) {
+        if (!reVersion.test(value)) {
+            return `${value} is not a valid version number. It must follow the format ${reVersion.source}`;
+        }
+        return true;
+    },
+
+    validateNotEmpty(value) {
+        if (!value.trim()) {
+            return `This value is mandatory!`;
+        }
+        return true;
+    }
```

Comment:
**comment:** I find these validator returns a little odd - returning a string or `true`. However, the use case is simple enough, so I suggest to leave it for now (unless you really feel like returning 2 values in a structure).

Reply:
Totally agree, but this is how [Hygen](https://www.hygen.io/) works. The other possibility would be to not use the prompt shorthand notation and create more complex prompts, like shown here: https://www.hygen.io/docs/generators#advanced-interactive-prompt